lib/shelley/exe/shelley-test-cluster.hs

```diff
@@ -0,0 +1,242 @@
+--
+--     [ "struggle", "section", "scissors", "siren"
+--     , "garbage", "yellow", "maximum", "finger"
+--     , "duty", "require", "mule", "earn"
+--     ]
+--
+-- - (Ledger) Created via the Ledger method for master key generation
+--
+--     [ "vague" , "wrist" , "poet" , "crazy" , "danger" , "dinner"
+--     , "grace" , "home" , "naive" , "unfold" , "april" , "exile"
+--     , "relief" , "rifle" , "ranch" , "tone" , "betray" , "wrong"
+--     ]
+--
+-- # CONFIGURATION
+--
+-- There are several environment variables that can be set to make debugging
+-- easier if needed:
+--
+-- - CARDANO_WALLET_PORT: choose a port for the API to listen on (default: random)
+--
+-- - CARDANO_NODE_TRACING_MIN_SEVERITY: increase or decrease the logging
+--                                      severity of the nodes. (default: Info)
+--
+-- - CARDANO_WALLET_TRACING_MIN_SEVERITY: increase or decrease the logging
+--                                        severity of cardano-wallet. (default: Info)
+--
+-- - TESTS_TRACING_MIN_SEVERITY: increase or decrease the logging severity of
+--                               the test cluster framework. (default: Notice)
+--
+-- - NO_POOLS: If set, the cluster will only start a BFT leader and a relay, no
+--             stake pools. This can be used for running test scenarios which do
+--             not require delegation-specific features without paying the
+--             startup cost of creating and funding pools.
+--
+-- - NO_CLEANUP: If set, the temporary directory used as a state directory for
+--               nodes and wallet data won't be cleaned up.
+main :: IO ()
+main = do
+    hSetBuffering stdout LineBuffering
+
+    nodeMinSeverity    <- nodeMinSeverityFromEnv
+    walletMinSeverity  <- walletMinSeverityFromEnv
+    clusterMinSeverity <- testMinSeverityFromEnv
+
+    let walletLogs =
+            [ LogToStdout walletMinSeverity
+            ]
+
+    let clusterLogs =
+            [ LogToStdout clusterMinSeverity
+            ]
+
+    withUtf8Encoding
+        $ withLoggingNamed "cardano-wallet" walletLogs
+        $ \(_, trWallet) -> withLoggingNamed "test-cluster" clusterLogs
+        $ \(_, trCluster) -> withSystemTempDir (trMessageText trCluster) "testCluster"
+        $ \dir -> withTempDir (trMessageText trCluster) dir "wallets"
+        $ \db -> withCluster
+            (trMessageText trCluster)
+            nodeMinSeverity
+            defaultPoolConfigs
+            dir
+            Nothing
+            whenByron
+            (whenShelley dir)
+            (whenReady trWallet trCluster db)
+  where
+    whenByron _ = pure ()
+
+    whenShelley dir _ = do
+        let encodeAddr = T.unpack . encodeAddress @'Mainnet
+        let addresses = map (first encodeAddr) shelleyIntegrationTestFunds
+        let accts = concatMap genRewardAccounts mirMnemonics
+        let rewards = (,Coin $ fromIntegral oneMillionAda) <$> accts
+        sendFaucetFundsTo stdoutTextTracer dir addresses
```

Comment:
We should use `trCluster` instead of `stdoutTextTracer` here! (Was changed recently in cardano-wallet:integration) Currently we see a huge about of logs about faucet txs when running `shelley-test-cluster`.